test/xgboost/test_experiment_config.py

```diff
@@ -0,0 +1,116 @@
+import pytest
+import copy
+
+from sklearn import datasets
+from sklearn.model_selection import train_test_split
+import xgboost as xgb
+import numpy as np
+
+from sigopt.xgboost.experiment import XGBExperiment
+
+iris = datasets.load_iris()
+X = iris.data
+y = iris.target
+num_class = 3
+X_train, X_test, Y_train, Y_test = train_test_split(X, y, test_size=0.2)
+D_train = xgb.DMatrix(X_train, label=Y_train)
+D_test = xgb.DMatrix(X_test, label=Y_test)
```

Comment:
I think we can achieve these with just Mock, so we don't actually need to import anything from sklearn.